### PR TITLE
change search-create name/filter options/parameters to match search UI 

### LINKED ISSUE #884: Update `search-create` CLI and python API commands to match `search`
**Expected behavior**
UI of `planet data search-create` and `planet data search` should be similar.

**Actual behavior (describe the problem)**
`planet data search` has filter and name as options, `planet data search-create` has name and filter as arguments.

**Proposed Solution**

Update `search-create` CLI and python API commands to match `search`.

### PR DESCRIPTION
**Pull request by jreiberkyle**
**Related Issue(s):**

Closes #884 

**Proposed Changes:**

For inclusion in changelog (if applicable):

1. Breaking change: Bring the UI of `planet data search-create` and `DataClient.create_search()` inline with `planet data search` and `DataClient.search()` by moving name and filter to required options in the CLI and rearranging the order of the parameters to `(item_types, search_filter, name)` in the Python library.

Not intended for changelog:

1. 

**Diff of User Interface**

Old behavior:

```console
planet data search-create name PSScene filter.json
```

```python
dataclient.create_search(name, ['PSScene'], search_filter)
```

New behavior:

```console
planet data search-create PSScene --filter filter.json --name search_name
```

```python
dataclient.create_search(['PSScene'],  search_filter=search_filter, name=name)
```

**PR Checklist:**

- [x] This PR is as small and focused as possible
- [x] If this PR includes proposed changes for inclusion in the changelog, the title of this PR summarizes those changes and is ready for inclusion in the Changelog.
- [] I have updated docstrings for function changes and docs in the 'docs' folder for user interface / behavior changes
- [x] This PR does not break any examples or I have updated them

**(Optional) @mentions for Notifications:**
There are no docs for these functions so there was nothing to update. #885 addresses adding them.

### CODE REVIEW

**Comment by cholmes** [comment]
Hrm, just realized the `search-update` should also be updated for consistency. Elsewhere (#887) I mentioned `planet data stats` could also use an update. I think that one is 'nice to have', but probably not essential. But `search-update` seems like it really should have the same behavior. I'm only trying it out now - I had thought it might just take JSON instead of having arguments, but just realized it has the arguments, and it'd be weird to have them different.

File: tests/integration/test_data_cli.py [hunk 1]
```diff
@@ -599,7 +605,12 @@ def test_data_search_create_filter_success(invoke, item_types):
     respx.post(TEST_SEARCHES_URL).return_value = mock_resp
 
     runner = CliRunner()
-    result = invoke(["search-create", name, item_types, json.dumps(filter)],
+    result = invoke([
+        "search-create",
+        item_types,
+        f'--name={name}',
+        f'--filter={json.dumps(filter)}'
```
Comment (kevinlacaille) [inline tests/integration/test_data_cli.py:612]: A little nit, but for consistency with the function's input, you could change the order of inputs to `item_types, search_filter, name`, which you already do for the test `test_data_search_create_daily_email()`
```suggestion
        item_types,
        f'--filter={json.dumps(filter)}',
        f'--name={name}'
```

**Review by kevinlacaille — APPROVED** [review]
All looks great, just a few small, very minor comments.

File: planet/cli/data.py [hunk 2]
```diff
@@ -321,11 +322,19 @@ async def search(ctx, item_types, filter, limit, name, sort, pretty):
...
+@click.option('--name',
+              type=str,
+              required=True,
+              help='Name of the saved search.')
+@click.option(
+    '--filter',
+    type=types.JSON(),
+    required=True,
+    help="""Filter to apply to search. Can be a json string, filename,
+              # or '-' for stdin.""")
```
[the remark this answers is not in the document]
Comment (jreiberkyle) [reply planet/cli/data.py:337]: yeah that's a typo good catch!

File: planet/cli/data.py [hunk 3]
```diff
@@ -336,12 +345,7 @@ async def search_create(ctx, name, item_types, filter, daily_email, pretty):
     This function outputs a full JSON description of the created search,
```
[the remark this answers is not in the document]
Comment (jreiberkyle) [reply planet/cli/data.py:345]: sure!

**Comment by kevinlacaille** [comment]
Awesome work, Jen!

**Pull request merged**